### PR TITLE
Update ledger

### PR DESCRIPTION
**Pull request by Tublad**
add class IntroducedOffer description for Subscription

### CODE REVIEW

File: Sources/Ledger/Classes/Product.swift [hunk 1]
```diff
@@ -9,15 +9,16 @@ public final class Product: CustomStringConvertible {
     public let identifier: String
     public let title: String
     public let info: String
-    public let price: String?
+    public let price: String
```
Comment (disabled) [inline Sources/Ledger/Classes/Product.swift]: Why this change? It's much easier to handle a missing price using an optional instead of some arbitrary string.

File: Sources/Ledger/Classes/Product.swift [hunk 2]
```diff
@@ -9,15 +9,16 @@ public final class Product: CustomStringConvertible {
...
+    public let period: SKProductSubscriptionPeriod.Period
```
Comment (disabled) [inline Sources/Ledger/Classes/Product.swift]: The same is for period. One-time purchases won't have a period, so it makes sense for it to be nil, instead of being `.unknown` - it's not unknown which period it is, there's just none.

File: Sources/Ledger/Extension/SKProductSubscriptionPeriod+Period.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            case let .month(count):
+               return descriptionPeriod("month", count: count)
+            case let .year(count):
+               return descriptionPeriod("year", count: count)
+            case .unknown:
+                return "N/A"
+            }
+        }
+
+        public func descriptionWithFormate(_ dateFormate: String = "d MMMM, yyyy") -> String {
```
Comment (disabled) [inline Sources/Ledger/Extension/SKProductSubscriptionPeriod+Period.swift]:
```suggestion
        public func descriptionWithFormat(_ dateFormat: String = "d MMMM, yyyy") -> String {
```

**Review by disabled — CHANGES_REQUESTED** [review]

**Review by stason314 — APPROVED** [review]

**Review by disabled — APPROVED** [review]
